Cabal/src/Distribution/Simple/GHC/Build/ExtraSources.hs

```diff
@@ -211,14 +226,12 @@ buildExtraSources
               sharedSrcOpts =
                 vanillaSrcOpts
                   `mappend` mempty
-                    { ghcOptFPic = toFlag True
-                    , ghcOptDynLinkMode = toFlag GhcDynamicOnly
+                    { ghcOptDynLinkMode = toFlag GhcDynamicOnly
                     }
               profSharedSrcOpts =
                 vanillaSrcOpts
                   `mappend` mempty
                     { ghcOptProfilingMode = toFlag True
-                    , ghcOptFPic = toFlag True
```

Comment:
The change to fpic business is worrying, it should be loudly documented (including the changelog, probably).

Reply:
This is a duplicate of the `vanillaSrcOpts`

```haskell
              vanillaSrcOpts =
                -- -fPIC is used in case you are using the repl
                -- of a dynamically linked GHC
                baseSrcOpts{ghcOptFPic = toFlag True}
```